tsdb/querier.go

```diff
@@ -106,14 +109,29 @@ func (q *querier) lvals(qs []Querier, n string) ([]string, error) {
 }
 
 func (q *querier) Select(ms ...*labels.Matcher) (SeriesSet, error) {
+	if len(q.blocks) != 1 {
+		return q.SelectSorted(ms...)
+	}
+	// Sorting Head series is slow, and unneeded when only the
+	// Head is being queried. Sorting blocks is a noop.
+	return q.blocks[0].Select(ms...)
+}
+
+func (q *querier) SelectSorted(ms ...*labels.Matcher) (SeriesSet, error) {
 	if len(q.blocks) == 0 {
 		return EmptySeriesSet(), nil
 	}
```

Comment:
I feek like we could avoid this extra function if we pu the if len ==  1 there and return q.blocks[0].Select(ms...)

Reply:
Not quite, as the sorting has to happen another function down again.

Reply:
As we just have one block arent the postings sorted already?

Reply:
mmh neverming in the head block it is probaly not sorted

Reply:
Exactly